build.py

```diff
@@ -271,6 +243,34 @@ def install_qsharp_python_package(cwd, wheelhouse, interpreter):
     subprocess.run(command_args, check=True, text=True, cwd=cwd)
 
 
+# If any package fails to install when using a requirements file, the entire
+# process will fail with unpredicatable state of installed packages. To avoid
+# this, we install each package individually from the requirements file.
+#
+# The reason we allow failures is that tooling for integration tests may not
+# be available on all platforms, so we don't want to fail the build if we can't
+# run the tests. The CI will run the tests on the platforms where the tooling
+# is available giving us the confidence that the tests pass on those platforms.
```

Comment:
I may be missing something, but this sounds antithetical to having reliable tests? Which problem are we trying to solve?